projects/com.oracle.truffle.llvm.types/src/com/oracle/truffle/llvm/types/LLVMFunction.java

```diff
@@ -216,4 +289,57 @@ public boolean equals(Object obj) {
         }
     }
 
+    private static class ForeignCallNode extends RootNode {
```

Comment:
I believe we should pass a [factory facade](https://github.com/graalvm/sulong/blob/master/projects/com.oracle.truffle.llvm.parser/src/com/oracle/truffle/llvm/parser/NodeFactoryFacade.java) to each `LLVMFunction`. The `LLVMFunction` can then use a newly introduced method to instantiate a `ForeignCallNode` (maybe better: `LLVMForeignCallNode`) that is implemented in `com.oracle.truffle.llvm.nodes.impl`. We could address that in a separate PR.